rllib/utils/actor_manager.py

```diff
@@ -376,6 +380,16 @@ def set_actor_state(self, actor_id: int, healthy: bool) -> None:
         """
         if actor_id not in self.__remote_actor_states:
             raise ValueError(f"Unknown actor id: {actor_id}")
+
+        was_healthy = self.__remote_actor_states[actor_id].is_healthy
```

Comment:
This was actually a bug:
The `Algorithm.on_workers_recreated` callback would NOT fire in case an actor got restarted b/c of another remote request made to it (other than the `ping()` in this method here, which called once per training iteration).